File: app/src/main/java/dev/ragnarok/fenrir/mvp/view/IFeedView.java
package dev.ragnarok.fenrir.mvp.view;

import androidx.annotation.Nullable;

import java.util.List;

import dev.ragnarok.fenrir.model.FeedSource;
import dev.ragnarok.fenrir.model.LoadMoreState;
import dev.ragnarok.fenrir.model.News;
import dev.ragnarok.fenrir.mvp.core.IMvpView;
import dev.ragnarok.fenrir.mvp.view.base.IAccountDependencyView;

public interface IFeedView extends IAccountDependencyView, IAttachmentsPlacesView, IMvpView, IErrorView {

    void displayFeedSources(List<FeedSource> sources);

    void notifyFeedSourcesChanged();

    void displayFeed(List<News> data, @Nullable String rawScrollState);

    void notifyFeedDataChanged();

    void notifyDataAdded(int position, int count);

    void notifyItemChanged(int position);

    void setupLoadMoreFooter(@LoadMoreState int state);

    void showRefreshing(boolean refreshing);

    void scrollFeedSourcesToPosition(int position);

    void scrollTo(int pos);

    void goToLikes(int accountId, String type, int ownerId, int id);

    void goToReposts(int accountId, String type, int ownerId, int id);

    void goToPostComments(int accountId, int postId, int ownerId);

    void showSuccessToast();

    void askToReload();
}

File: app/src/main/java/dev/ragnarok/fenrir/mvp/view/IFeedView.kt
package dev.ragnarok.fenrir.mvp.view

import dev.ragnarok.fenrir.model.FeedSource
import dev.ragnarok.fenrir.model.LoadMoreState
import dev.ragnarok.fenrir.model.News
import dev.ragnarok.fenrir.mvp.core.IMvpView
import dev.ragnarok.fenrir.mvp.view.base.IAccountDependencyView

interface IFeedView : IAccountDependencyView, IAttachmentsPlacesView, IMvpView, IErrorView {
    fun displayFeedSources(sources: MutableList<FeedSource>)
    fun notifyFeedSourcesChanged()
    fun displayFeed(data: MutableList<News>, rawScrollState: String?)
    fun notifyFeedDataChanged()
    fun notifyDataAdded(position: Int, count: Int)
    fun notifyItemChanged(position: Int)
    fun setupLoadMoreFooter(@LoadMoreState state: Int)
    fun showRefreshing(refreshing: Boolean)
    fun scrollFeedSourcesToPosition(position: Int)
    fun scrollTo(pos: Int)
    override fun goToLikes(accountId: Int, type: String?, ownerId: Int, id: Int)
    override fun goToReposts(accountId: Int, type: String?, ownerId: Int, id: Int)
    fun goToPostComments(accountId: Int, postId: Int, ownerId: Int)
    fun showSuccessToast()
    fun askToReload()
}